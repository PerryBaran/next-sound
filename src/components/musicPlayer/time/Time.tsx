import style from "./time.module.css"

export default function Time(props: { time: number }) {
  const { time } = props

  const displayTime = (seconds: number) => {
    let sec: number | string = seconds
    let min = 0
    while (sec >= 60) {
      min++
      sec = sec - 60
    }
    if (sec < 10) {
      sec = `0${sec}`
    }
    return `${min}:${sec}`
  }

  return (
    <div className={style.container}>
      <p>{displayTime(time)}</p>
    </div>
  )
}
