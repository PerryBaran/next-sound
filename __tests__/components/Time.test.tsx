import { render, screen } from "@testing-library/react"
import Time from "../../src/components/musicPlayer/time/Time"

describe("Time", () => {
  test("renders correctly when time is less than 10", () => {
    render(<Time time={9} />)

    expect(screen.getByText("0:09")).toBeInTheDocument()
  })

  test("renders correctly when time is greater than 10", () => {
    render(<Time time={11} />)

    expect(screen.getByText("0:11")).toBeInTheDocument()
  })

  test("renders correctly when time is greater than 60", () => {
    render(<Time time={61} />)

    expect(screen.getByText("1:01")).toBeInTheDocument()
  })

  test("renders correctly when time is greater than 600", () => {
    render(<Time time={601} />)

    expect(screen.getByText("10:01")).toBeInTheDocument()
  })

  test("snapshot", () => {
    const { asFragment } = render(<Time time={601} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
