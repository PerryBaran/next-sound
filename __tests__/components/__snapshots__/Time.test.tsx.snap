// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p>
      10:01
    </p>
  </div>
</DocumentFragment>
`;
